builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -22,7 +22,8 @@ func resourceCloudStackInstance() *schema.Resource {
 		Schema: map[string]*schema.Schema{
 			"name": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+				Computed: true,
 				ForceNew: true,
```

Comment:
`ForceNew: true` needs to be removed, since you added logic to update the `name` of existing instances